src/helper/PendlePTRateProvider.sol

```diff
@@ -0,0 +1,35 @@
+pragma solidity 0.8.21;
+
+import { IRateProvider } from "src/interfaces/IRateProvider.sol";
+
+import { IPMarket } from "lib/ion-protocol/lib/pendle-core-v2-public/contracts/interfaces/IPMarket.sol";
+
+import { IStandardizedYield } from "lib/ion-protocol/lib/pendle-core-v2-public/contracts/interfaces/IStandardizedYield.sol";
+import { IPPtLpOracle } from "lib/ion-protocol/lib/pendle-core-v2-public/contracts/interfaces/IPPtLpOracle.sol";
+
+/**
+ * @title PendlePTRateProvider
+ * @custom:security-contact security@molecularlabs.io
+ */
+contract PendlePTRateProvider is IRateProvider {
+    /// @notice constant values
+    IPPtLpOracle public constant ORACLE = IPPtLpOracle(0x14030836AEc15B2ad48bB097bd57032559339c92);
+    uint32 public constant DURATION = 1 days;
+
+    /// @notice the pendle market this rate provider serves
+    IPMarket public immutable market;
+
+    /// @param pendleMarket to serve the PT rate
+    constructor(IPMarket pendleMarket) {
+        market = pendleMarket;
+    }
```

Comment:
duration should be in the constructor and immutable for this contract to be reusable across multiple durations